routes/dao_coin_exchange_with_fees.go

```diff
@@ -178,33 +196,70 @@ func (fes *APIServer) UpdateDaoCoinMarketFees(ww http.ResponseWriter, req *http.
+	}
+	associations, err := utxoView.GetUserAssociationsByAttributes(associationFilter)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("UpdateDaoCoinMarketFees: Problem fetching associations: %v", err))
+		return
+	}
+	if len(associations) == 1 {
+		association := associations[0]
+		if IsExtraDataValueTrue(association.ExtraData, lib.DisableTradingFeeUpdateKey) {
+			_AddBadRequestError(ww, fmt.Sprintf(
+				"UpdateDaoCoinMarketFees: Trading fee updates are disabled for this market"))
+			return
+		}
+	} else if len(associations) != 0 {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"UpdateDaoCoinMarketFees: Expected at most one association but got %d", len(associations)))
+		return
+	}
+
 	// Compute the additional transaction fees as specified by the request body and the node-level fees.
 	additionalOutputs, err := fes.getTransactionFee(
-		lib.TxnTypeUpdateProfile, updaterPublicKeyBytes, requestData.TransactionFees)
+		lib.TxnTypeUpdateProfile, ammPublicKeyBytes, requestData.TransactionFees)
 	if err != nil {
 		_AddBadRequestError(ww, fmt.Sprintf("AuthorizeDerivedKey: TransactionFees specified in Request body are invalid: %v", err))
 		return
 	}
 
-	// Try and create the UpdateProfile txn for the user.
-	txn, totalInput, changeAmount, fees, err := fes.blockchain.CreateUpdateProfileTxn(
-		updaterPublicKeyBytes,
-		profilePublicKeyBytes,
-		"", // Don't update username
-		"", // Don't update description
-		"", // Don't update profile pic
-		existingProfileEntry.CreatorCoinEntry.CreatorBasisPoints, // Don't update creator basis points
-		// StakeMultipleBasisPoints is a deprecated field that we don't use anywhere and will delete soon.
-		// I noticed we use a hardcoded value of 12500 in the frontend and when creating a post so I'm doing
-		// the same here for now.
-		1.25*100*100,                  // Don't update stake multiple basis points
-		existingProfileEntry.IsHidden, // Don't update hidden status
-		0,                             // Don't add additionalFees
-		additionalExtraData,           // The new ExtraData
-		requestData.MinFeeRateNanosPerKB, fes.backendServer.GetMempool(),
-		additionalOutputs)
+	// Try and create the association txn for the UpdaterPublicKey.
+	txn, totalInput, changeAmount, fees, err := fes.blockchain.CreateCreateUserAssociationTxn(
+		ammPublicKeyBytes,
+		&lib.CreateUserAssociationMetadata{
+			TargetUserPublicKey: lib.NewPublicKey(profilePublicKeyBytes),
+			AppPublicKey:        lib.NewPublicKey(ammPublicKeyBytes),
+			AssociationType:     []byte(lib.TokenTradingFeesByPkidMapKey),
+			AssociationValue:    []byte(lib.TokenTradingFeesByPkidMapKey),
+		},
+		additionalExtraData,
+		requestData.MinFeeRateNanosPerKB,
+		fes.backendServer.GetMempool(),
+		additionalOutputs,
+	)
 	if err != nil {
-		_AddBadRequestError(ww, fmt.Sprintf("UpdateDaoCoinMarketFees: Problem creating transaction: %v", err))
+		_AddInternalServerError(ww, fmt.Sprintf("CreateUserAssociation: problem creating txn: %v", err))
```

Comment:
Let's make this error more clear. I think it should be `UpdateDaoCoinMarketFees: problem creating user association txn: %v`